version.go

```diff
@@ -17,6 +17,15 @@ var (
 	gitTreeState = ""                     // determined from `git status --porcelain`. either 'clean' or 'dirty'
 )
 
+// ImageTag return the image tag.
+// GetVersion().Version adulterates the version making it useless as the image tag.
+func ImageTag() string {
+	if version != "untagged" {
```

Comment:
This should be using `GetVersion()` otherwise it's always `"v0.0.0"`?

Reply:
Nope. The value of `version` and other vars is set by build flags in the `Makefile`:

```
GIT_TAG               := $(shell git describe --exact-match --tags --abbrev=0  2> /dev/null || echo untagged)
...
VERSION               := latest
...
ifeq ($(RELEASE_TAG),true)
VERSION               := $(GIT_TAG)
endif
...
override LDFLAGS += \
  -X github.com/argoproj/argo-workflows/v3.version=$(VERSION) \
  -X github.com/argoproj/argo-workflows/v3.buildDate=${BUILD_DATE} \
  -X github.com/argoproj/argo-workflows/v3.gitCommit=${GIT_COMMIT} \
  -X github.com/argoproj/argo-workflows/v3.gitTreeState=${GIT_TREE_STATE}
...
go build -v -ldflags '${LDFLAGS} -extldflags -static' -o $@.exe ./cmd/argo
```

Reply:
Learn more: https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications